classpractice/full-name/src/test/java/org/karolina/FullNameTest.java

```diff
@@ -0,0 +1,141 @@
+package org.karolina;
+
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class FullNameTest {
+
+    @ParameterizedTest
+    @MethodSource ("testCasesWithNullsInNotAllowedPlaces")
+    public void AllArgumentsInOverloadedConstructorsAreNotNull (Executable executable) {
+        var calledConstructor = executable;
+
+        assertThrows(NullPointerException.class, calledConstructor);
+    }
+
+    public void ifPassedMiddleNameIsNullItIsConvertedToEmptyString () {
+        var fullName = new FullName("abc", null, "def", "ghi");
+
+        var extractedMiddleName = fullName.getMiddleName();
+
+        assertEquals("", extractedMiddleName);
```

Comment:
trochę nieoczekiwane API - zakładałbym, że jak nie ma wartości, to będzie `null`, a nie `""`